Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -222,9 +241,23 @@ protected override void ProcessRecord()
 
             // Content type to use for the new object.
             string objContentType = null;
-
             Stream contentStream = null;
-            if (!string.IsNullOrEmpty(File))
+
+            if (ParameterSetName == ParameterSetNames.ContentsFromFolder)
+            {
+                // User gives us the path to a folder, we will resolve the path and upload the contents of that folder.
+                string resolvedFolderPath = GetFullPath(Folder);
+                if (string.IsNullOrWhiteSpace(resolvedFolderPath) || !Directory.Exists(resolvedFolderPath))
+                {
+                    throw new DirectoryNotFoundException($"Directory '{resolvedFolderPath}' cannot be found.");
+                }
+
+                // The relative directory name of this directory will be the name of the directory itself.
+                UploadDirectory(resolvedFolderPath, metadataDict, Path.GetFileName(resolvedFolderPath));
```

Comment:
I don't like the complexity of adding yet another parameter, but it does seem wrong to always upload a folder into the root of the GCS bucket.

Why don't we add a new parameter here:

````
/// When uploading the contents of a directory into Google Cloud Storage, this is the prefix applied
/// to every object which is uploaded.
public string ObjectNamePrefix { get; set; } = "";
````

And then in this call to `UploadDirectory` we just prefix `Path.GetFileName` with the parameter's value.